packages/cli-core/src/printer.ts

```diff
@@ -0,0 +1,7 @@
+import { LogLevel, Printer } from './printer/printer.js';
+
+const minimumLogLevel = process.argv.includes('--debug')
+  ? LogLevel.DEBUG
+  : LogLevel.INFO;
+
+export const printer = new Printer(minimumLogLevel);
```

Comment:
We should either
1. Inject the printer from cli as needed into cli-core
2. Depend on this printer instance in cli and create-amplify rather than creating instances in those packages

Otherwise we are going to have multiple instances of the printer which is supposed to be a singleton. Option 2 is probably easier but option 1 is the "most correct" in terms of dependency inversion. However, we are already using the pattern of importing the printer across the cli package rather than injecting it, so perhaps option 2 is the way to go.

@sobolk thoughts?

also see https://github.com/aws-amplify/amplify-backend/pull/903/files

Reply:
I think this https://github.com/aws-amplify/amplify-backend/pull/947#discussion_r1465546808 could be a solution for this too.
If package manager controller is type in platform types but it's factory is in cli-core then that factory may accept printer instance in some way.

Reply:
Yeah that would work. That would basically be option 1

Reply:
We need to replace the printer instances [here](https://github.com/aws-amplify/amplify-backend/blob/main/packages/cli/src/printer.ts) and [here](https://github.com/aws-amplify/amplify-backend/blob/main/packages/create-amplify/src/printer.ts) with this instance. This can go in a follow up PR if you want.

We should also consider making Printer a type in `@aws-amplify/plugin-types` and only export an instance from `@aws-amplify/cli-core` rather than the class. This would make testing more annoying so we could also consider a lint rule to forbid new instances of Printer. Probably better to cut a GH issue to track this than do it all in this PR.

Reply:
Good catch. issue cut https://github.com/aws-amplify/amplify-backend/issues/958